### PR TITLE
feat: Allow BeginDialog to use arbitrary dialog id

### LINKED ISSUE #5820: BeginDialog.Dialog should allow you type in a arbitrary dialog.id.
## Describe the bug
BeginDialog does not let me begin a dialog by id, it only shows a list of adaptive dialogs.  

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## Expected behavior

**BeginDialog** action is designed to pass options to 
* An **AdaptiveDialog** by **resourceId**
* A **"legacy" dialog** by the **dialog.id** a dialog is registered.
The LegacyDialog is added to a a DialogContainer in the parent (such as AdaptiveDialog or root DialogManager).  When it is added it is given a unique id by the developer adding it, which can then be used to invoke it with **BeginDialog** action

startup.cs
```C#
   DialogManager dm = new DialogManager();
   dm.Dialogs.Add(new FooDialog(...) { Id = "DoFoo3", ....}
```

MyDialog.dialog
```json
    {
       "$kind":"Microsoft.BeginDialog",
        "dialog":"DoFoo3"
         "options": {
              "name":"=user.name"
         }
     }
```
It is fine to have a drop down of the known AdaptiveDialogs, but the UI should let me type in an dialog id or expression to a dialog id. **If we don't have this people customers cannot access dialogs they have written in code.** 
```json
    {
       "$kind":"Microsoft.BeginDialog",
        "dialog":"=conversation.somevar"
         "options": {
              "name":"=user.name"
         }
     }
```

## Screenshots

image

### PR DESCRIPTION
**Pull request by tdurnford**
## Description
Allows user to select 'Write Expression' from the select dialog dropdown and insert arbitrary dialog id.

## Task Item
Closes #5820

## Screenshots
Mar-01-2021 09-58-54

### CODE REVIEW

File: Composer/packages/ui-plugins/select-dialog/src/SelectDialog.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { IComboBoxOption, SelectableOptionMenuItemType } from 'office-ui-fabric-react/lib/ComboBox';
 import { FieldProps, useShellApi } from '@bfc/extension-client';
 import formatMessage from 'format-message';
+import { IntellisenseTextField } from '@bfc/adaptive-form';
+import { WithTypeIcons } from '@bfc/adaptive-form/lib/components/WithTypeIcons';
```
Comment (a-b-r-o-w-n) [inline Composer/packages/ui-plugins/select-dialog/src/SelectDialog.tsx:9]: This looks like it should be exported from the main module of `@bfc/adaptive-form`.

**Review by a-b-r-o-w-n — CHANGES_REQUESTED** [review]

**Comment by cwhitten** [comment]
@tdurnford can we change the icon from the `Write an Expression` option? It looks like another Dialog and confuses me just a bit. @sangwoohaan what do you think?

**Comment by tdurnford** [comment]
@cwhitten I changed the icon to an equal sign in the updated expression icon commit

File: Composer/packages/ui-plugins/select-dialog/src/ComboBoxField.tsx [hunk 2]
```diff
@@ -36,7 +36,9 @@ export const ComboBoxField: React.FC<ComboBoxFieldProps> = (props) => {
       <div>
         <Icon
           aria-hidden="true"
-          iconName={option.key === ADD_DIALOG ? 'Add' : 'OpenSource'}
+          iconName={
+            option.key === ADD_DIALOG ? 'Add' : option.key === 'expression' ? 'CalculatorEqualTo' : 'OpenSource'
```
[the remark this answers is not in the document]
Comment (tdurnford) [reply Composer/packages/ui-plugins/select-dialog/src/ComboBoxField.tsx:40]: 😆

**Review by cwhitten — APPROVED** [review]

**Pull request merged**